kong/partial_service_test.go

```diff
@@ -0,0 +1,455 @@
+	})
+
+	t.Run("invalid partial creation -  empty partial", func(_ *testing.T) {
+		notCreatedPartial, err := client.Partials.Create(defaultCtx, &Partial{})
+
+		assert.Error(err)
+		assert.ErrorContains(err, "partial type cannot be nil")
+		assert.Nil(notCreatedPartial)
+	})
+
+	t.Run("invalid partial creation -  nil type", func(_ *testing.T) {
+		notCreatedPartial, err := client.Partials.Create(defaultCtx, &Partial{
+			Name: String("my-test-partial"),
+		})
+
+		assert.Error(err)
+		assert.ErrorContains(err, "partial type cannot be nil")
+		assert.Nil(notCreatedPartial)
+	})
+
+	t.Run("create valid partial", func(_ *testing.T) {
+		validPartial := &Partial{
+			Name: String("my-test-partial"),
+			Type: String("redis-ee"),
+		}
+
+		createdPartial, err := client.Partials.Create(defaultCtx, validPartial)
+		require.NoError(err)
+		require.NotNil(createdPartial)
+		assert.Equal("my-test-partial", *createdPartial.Name)
+		assert.Equal("redis-ee", *createdPartial.Type)
+		assert.Equal(defaultConfigRedisEEPartial, createdPartial.Config)
+
+		t.Cleanup(func() {
+			if createdPartial != nil {
+				assert.NoError(client.Partials.Delete(defaultCtx, createdPartial.ID))
+			}
+		})
+	})
+}
+
+func TestPartialServiceGetEndpoint(t *testing.T) {
+	RunWhenEnterprise(t, ">=3.10.0", RequiredFeatures{})
+
+	require := require.New(t)
+	assert := assert.New(t)
+
+	client, err := NewTestClient(nil, nil)
+	require.NoError(err)
+	require.NotNil(client)
+
+	t.Run("invalid get -  empty id", func(_ *testing.T) {
+		noPartial, err := client.Partials.Get(defaultCtx, String(""))
+
+		assert.Error(err)
+		assert.ErrorContains(err, "partialID cannot be nil for Get operation")
+		assert.Nil(noPartial)
+	})
+
+	t.Run("get by id", func(_ *testing.T) {
+		createdPartial, err := client.Partials.Create(defaultCtx, &Partial{
+			Name: String("my-test-partial"),
+			Type: String("redis-ee"),
+		})
+		require.NoError(err)
+
+		t.Cleanup(func() {
+			if createdPartial != nil {
+				assert.NoError(client.Partials.Delete(defaultCtx, createdPartial.ID))
+			}
+		})
+
+		fetchedPartial, err := client.Partials.Get(defaultCtx, createdPartial.ID)
+		assert.NoError(err)
+		assert.NotNil(fetchedPartial)
+		assert.Equal("my-test-partial", *fetchedPartial.Name)
+		assert.Equal("redis-ee", *fetchedPartial.Type)
+	})
+
+	// t.Run("get by name", func(_ *testing.T) {
+	// 	createdPartial, err := client.Partials.Create(defaultCtx, &Partial{
+	// 		Name: String("my-demo-partial"),
+	// 		Type: String("redis-ee"),
+	// 	})
+	// 	require.NoError(err)
+
+	// 	t.Cleanup(func() {
+	// 		if createdPartial != nil {
+	// 			assert.NoError(client.Partials.Delete(defaultCtx, createdPartial.ID))
+	// 		}
+	// 	})
+
+	// 	fetchedPartial, err := client.Partials.Get(defaultCtx, String("my-demo-partial"))
+	// 	assert.NoError(err)
+	// 	assert.NotNil(fetchedPartial)
+	// 	assert.Equal("my-demo-partial", *fetchedPartial.Name)
+	// 	assert.Equal("redis-ee", *fetchedPartial.Type)
+	// })
```

Comment:
Why is this commented out?

Reply:
Ohh I left it by mistake. I will remove it.
Earlier the APIs were supporting GET, DELETE by name as well. Later, the gateway team only allowed all operations by ID to keep things in sync with Konnect, I suppose. So, this test was added earlier but I later forgot to remove it.